misc/docs-linter/links.go

```diff
@@ -3,105 +3,73 @@ package main
 import (
 	"bufio"
 	"bytes"
-	"context"
 	"fmt"
-	"golang.org/x/sync/errgroup"
-	"io"
-	"mvdan.cc/xurls/v2"
-	"net/http"
+	"os"
+	"path/filepath"
+	"regexp"
 	"strings"
-	"sync"
 )
 
-// extractUrls extracts URLs from a file and maps them to the file
-func extractUrls(fileContent []byte) []string {
+// extractLocalLinks extracts links to local files given file content
+func extractLocalLinks(fileContent []byte) []string {
 	scanner := bufio.NewScanner(bytes.NewReader(fileContent))
-	urls := make([]string, 0)
+	links := make([]string, 0)
+	// Regular expression to match markdown links
+	re := regexp.MustCompile(`]\((\.\.?/.+?)\)`)
 
 	// Scan file line by line
 	for scanner.Scan() {
 		line := scanner.Text()
 
-		// Extract links
-		rxStrict := xurls.Strict()
-		url := rxStrict.FindString(line)
+		// Find all matches
+		matches := re.FindAllString(line, -1)
 
-		// Check for empty links and skip them
-		if url == " " || len(url) == 0 {
-			continue
-		}
+		// Extract and print the local file links
+		for _, match := range matches {
+			// Remove ]( from the beginning and ) from end of link
+			match = match[2 : len(match)-1]
+
+			// Remove markdown headers in links
+			pos := strings.Index(match, "#")
+
+			if pos != -1 {
```

Comment:
Minor suggestion: for situations like this where there is an assignment then a conditional, I believe it to be cleaner if the two are combined -- 
```go
if pos := strings.Index(match, "#"); pos != 1 {
    match = match[:pos]
}
```

The same applies to next assignment and conditional.

Reply:
[66758ef](https://github.com/gnolang/gno/pull/2416/commits/66758ef3fdb86b3b1e201fd8ce6c864e18737cd4)